tests/example_http_key_value_store.rs

```diff
@@ -0,0 +1,44 @@
+mod test_server;
+
+use http::StatusCode;
+use rama::{error::BoxError, http::Request};
+
+use crate::test_server::recive_as_string;
+
+#[tokio::test]
+async fn test_http_key_value_store() -> Result<(), BoxError> {
+    let _example = test_server::run_example_server("http_key_value_store");
+
+    // because the example couldn't recive request yet.
```

Comment:
Probably not a bad idea to just make a high level test client under `test_client.rs` of this crate which gives you easy methods to get `json/str` data, and which will also have graceful retries. E.g. if a failure to retry with a backoff policy. The latter could be made using a new middleware that could be added under `rama::service::layers`. Called `RetryService` / `RetryLayer`. Will look very much like the `Limit` stuff, but then for retrying. Also can be used using a policy,  so that for example exponential backoff is possible here as well, etc.

I get that this is asking a lot though, on top of the hard work you're putting with this request. So feel free to skip it for now and I can revisit that later. I'm only saying it because right now you are adding methods, e.g. `receive_as_string` and also small delays etc... That's a lot of sprinkled choices that in reality you can better solve by providing a small high level client for testing purposes.

But again, feel free to skip. Or come back to me if you do want to do it but need some guidance. We can also have it as a separate PR and keep this one scoped , a lot of possibilities. You can let me know.

Reply:
Added the stories:

* https://github.com/plabayo/rama/issues/138
* https://github.com/plabayo/rama/issues/137

Do not feel obliged to pick these up, or do it as part of this PR. It can be done by someone else, and it can be also done on another time. All up to you, either way fine.

Reply:
Thank you for your concern.
It is difficult for me to create high level client. I would skip it in this PR.

Reply:
That is fair enough. You are already doing a lot! Thx for that!